geometric_sqrt.go

```diff
@@ -0,0 +1,51 @@
+package interpolator
+
+import (
+	"fmt"
+	"math"
+)
+
+// GeometricSqrt is a geometric interpolator scaled with a square root.
+type GeometricSqrt struct {
+	xys XYs
+}
+
+// NewGeometricSqrt builds a geometric sqrt interpolator.
+// The input `xys` must be ordered, have unique abscissas
+// and positive ordinates.
+func NewGeometricSqrt(xys XYs) (*GeometricSqrt, error) {
+	if l := len(xys); l < 2 {
+		return nil, fmt.Errorf("at least 2 points are required to build a geometric sqrt interpolator, but got %d", l)
+	}
+	for _, xy := range xys {
+		if xy.Y < epsilon {
+			return nil, fmt.Errorf("input xys must have non-negative ordinates")
+		}
+	}
+	return &GeometricSqrt{
+		xys: xys,
+	}, nil
+}
+
+// Value compute the value of f(x) based on geometric sqrt interpolation with threshold.
```

Comment:
here as well, explain a bit more what `with threshold` means (even in two lines)